kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/util/NetUtil.java

```diff
@@ -26,27 +27,45 @@ public static String hardwareAddressToString(byte[] macAddress) {
         if (macAddress == null) {
             return "N/A";
         }
-        StringBuilder sb = new StringBuilder();
-        for (int i = 0; i < macAddress.length; i++) {
-            sb.append(String.format("%02X%s", macAddress[i], i < macAddress.length - 1 ? ":" : ""));
+
+        if (macAddress.length != 6) {
+            throw new IllegalArgumentException("macAddress is invalid");
+        }
+
+        StringJoiner sj = new StringJoiner(":");
+        for (byte item : macAddress) {
+        	sj.add(String.format("%02X", item));
         }
-        return sb.toString();
+        
+        return sj.toString();
     }
 
     public static byte[] hardwareAddressToBytes(String macAddress) {
         if (macAddress == null || macAddress.isEmpty()) {
             return new byte[] { 0, 0, 0, 0, 0, 0 };
         }
+        
+        String[] items = macAddress.split("\\:");
 
-        macAddress = macAddress.replaceAll(":", "");
+        if (items.length != 6) {
+            throw new IllegalArgumentException("mac is invalid: " + macAddress);
```

Comment:
this is OK instead. The old code would have likely thrown an ArrayIndexOutOfBoundsException accessing characters past the end of the string